### ISSUE TITLE
train_on_label

### ISSUE THREAD

**Issue by Chacioc** [tloen/alpaca-lora#495]
How to set the "train_on_label" flag for predict tasks during fine-tuning

**Comment by ElleLeonne** [comment]
Keep in mind this is a decoder language model.

Cross-Entropy is a prediction loss already.

In order to only train on a single token, you need to set the "labels"  dict for every other token to -100.

There's an example of that in the "train on inputs" part of the code.

You might be used to having your classification token come at the start of the sequence, but since we're a decoder only model, you'll probably want to put at the end.

You may also need to write a new head for the model, if the CausalLM head doesn't suit your needs in this regard.
